test/runtime/runtime_test.go

```diff
@@ -22,61 +23,49 @@ var _ = Suite(&testingRuntimeSuite{})
 
 func (s *testingRuntimeSuite) TestPrependEnvsPrefix(c *C) {
 	m := []struct {
-		comment     string
-		cmd         string
-		env         map[string]string
-		soft        bool
-		expectedCmd string
-		err         string
+		comment      string
+		cmd          string
+		env          map[string]string
+		soft         bool
+		expectedCmd  string
+		expectedEnv  map[string]string
+		expectedSoft bool
+		err          string
 	}{
 		{
 			"no variable in environment",
-			"/node server.js",
-			map[string]string{},
-			false,
-			"/node server.js",
+			"/node server.js", map[string]string{}, false,
+			"/node server.js", map[string]string{}, false,
 			"",
 		},
 		{
 			"single variable in environment",
-			"/node server.js",
-			map[string]string{"PORT": "8000"},
-			false,
-			"--env=PORT=8000 /node server.js",
+			"/node server.js", map[string]string{"PORT": "8000"}, false,
+			"/node server.js", map[string]string{"PORT": "8000"}, false,
 			"",
 		},
 		{
 			"two variables in environment",
-			"/node server.js",
-			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
-			false,
-			// Order is not guaranteed hence the following regex is needed:
-			"(--env=PORT=8000 --env=ENDPOINT=foo.com|--env=ENDPOINT=foo.com --env=PORT=8000) /node server.js",
+			"/node server.js", map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"}, false,
```

Comment:
I understand now. So I've modified so that we use list instead of map. I like it how now test is more readable. Also, the checker is now more general (supports validating any `PREFIX_LIST + BASE` string). We should probably find a better name for it, but I can't think of one ATM. Any suggestion?